components/match2/commons/opponent.lua

```diff
@@ -278,30 +282,36 @@ Template:LiteralOpponent, and etc.
 Wikis sometimes provide variants of this function that include wiki specific
 transformations.
 ]]
-function Opponent.readOpponentArgs(args)
+function Opponent.readOpponentArgs(args, date)
 	local partySize = Opponent.partySize(args.type)
 
 	if args.type == Opponent.team then
 		local template = args.template or args[1]
 		return template and {
 			type = Opponent.team,
-			template = template:lower():gsub('_', ' '),
+			template = template:lower(),
 		}
 
 	elseif partySize == 1 then
 		local player = {
 			displayName = args[1] or args.p1 or args.name or '',
-			flag = String.nilIfEmpty(Flags.CountryName(args.flag)),
-			pageName = args.link,
+			flag = String.nilIfEmpty(Flags.CountryName(args.flag or args.p1flag)),
+			pageName = args.link or args.p1link,
+			team = args.team or args.p1team,
 		}
 		return {type = Opponent.solo, players = {player}}
 
 	elseif partySize then
-		local players = Array.map(Array.range(1, partySize), function(i)
+		local players = Array.map(Array.range(1, partySize), function(playerIndex)
+			local playerTeam = args['p' .. playerIndex .. 'team']
+			if playerTeam then
+				playerTeam = playerTeam:lower()
+			end
```

Comment:
is the `:lower()` check needed? wouldn't it get normalized by TeamTemplate.resolve anyway?

Reply:
it currently does not

Reply:
seems like `TeamTemplate.resolve` already normalizes cases so `:lower()` isn't needed